Slim/Web/Settings/Server/Behavior.pm

```diff
@@ -82,9 +84,78 @@ sub handler {
 		}
 
 		$prefs->set('releaseTypesToIgnore', [ keys %releaseTypesToIgnore ]);
+
+		foreach my $role (Slim::Schema::Contributor::defaultContributorRoles()) {
+			$prefs->set(lc($role)."AlbumLink", $paramRef->{"pref_".lc($role)."AlbumLink"} ? "1" : "0");
+			next if $role eq "ALBUMARTIST" || $role eq "ARTIST";
+			$prefs->set(lc($role)."InArtists", $paramRef->{"pref_".lc($role)."InArtists"} ? "1" : "0");
+		}
+		foreach my $role (Slim::Schema::Contributor::userDefinedRoles()) {
+			$userDefinedRoles->{$role}->{albumLink} = $paramRef->{"pref_".lc($role)."AlbumLink"};
+			$userDefinedRoles->{$role}->{include} = $paramRef->{"pref_".lc($role)."InArtists"};
+		}
```

Comment:
Just a style note: I'd prefer to have a bit more "air" in the code. Eg. in concatenations I prefer to have spaces around the dot. Just to make it clearer there's an operator. Or leave an empty line between multiline blocks. etc.

For the next PR 😉.